core/src/main/java/nl/nn/adapterframework/receivers/JavaListener.java

```diff
@@ -131,11 +131,18 @@ public RawMessageWrapper<M> wrapRawMessage(M rawMessage, PipeLineSession session
 	@Override
 	public String processRequest(String correlationId, String rawMessage, HashMap context) throws ListenerException {
 		try {
+			String messageId;
 			if (context != null) {
 				context.put(PipeLineSession.CORRELATION_ID_KEY, correlationId);
+				messageId = (String) context.get(PipeLineSession.MESSAGE_ID_KEY);
+			} else {
+				messageId = null;
+			}
+			RawMessageWrapper<M> rawMessageWrapper = new RawMessageWrapper<>((M)rawMessage, messageId, correlationId);
+			Message result;
+			try (PipeLineSession session = new PipeLineSession(context != null ? context : Collections.emptyMap())) {
```

Comment:
Create session here so we do not have to do it in the method we call, b/c that method in most cases already gets a session specific for this call.